src/software/ai/navigator/obstacle/robot_navigation_obstacle_factory.cpp

```diff
@@ -62,6 +62,20 @@ RobotNavigationObstacleFactory::createStaticObstaclesFromMotionConstraint(
             obstacles.push_back(createFromFieldRectangle(
                 field.enemyHalf(), field.fieldLines(), field.fieldBoundary()));
             break;
+        case TbotsProto::MotionConstraint::ENEMY_HALF_WITHOUT_CENTRE_CIRCLE:
+        {
+            double radius                        = field.centerCircleRadius() + 0.2;
+            Polygon centre_circle_and_enemy_half = Polygon(
+                {Point(0, field.fieldBoundary().yLength() / 2), Point(0, radius),
+                 Point(radius * std::cos(M_PI / 4), radius * std::sin(M_PI / 4)),
+                 Point(radius, 0),
+                 Point(radius * std::cos(M_PI / 4), -radius * std::sin(M_PI / 4)),
+                 Point(0, -radius), Point(0, -field.fieldBoundary().yLength() / 2),
+                 field.fieldBoundary().posXNegYCorner(),
+                 field.fieldBoundary().posXPosYCorner()});
+            obstacles.push_back(createFromShape(centre_circle_and_enemy_half));
+            break;
+        }
```

Comment:
I think for the shape, we want something that goes around the entirety of the center circle. If possible, the line from the half circle connecting the polygon to the edge of the field should also be vertical. Something maybe similar to the red line:
<img width="531" alt="image" src="https://user-images.githubusercontent.com/28585597/216810662-462f33a9-922a-43f0-8e8c-010496353766.png">

Reply:
Updated the shape to look like this:
![Screenshot from 2023-02-11 12-20-52](https://user-images.githubusercontent.com/47040320/218279657-80b9cc55-dfbf-4406-a960-3904b3e54f94.png)

I shrunk the radius so that our friendly kickoff robot is only allowed inside the whole centre circle.

Reply:
Discussed offline with Nima, will resize the centre circle obstacle cutout to match the actual centre circle to accommodate for HRVO

Reply:
Updated motion constraint for enemy half without centre circle
![image](https://user-images.githubusercontent.com/47040320/218953127-cc736652-a8e5-4766-bbb8-8f6a4c8bbd40.png)